trace/trace.go

```diff
@@ -31,6 +32,36 @@ type Field struct {
 	Value interface{}
 }
 
+type stepTraceTime interface {
+	time() time.Time
+}
+
+func (t *Trace) time() time.Time {
+	return t.startTime
+}
+
+func (s traceStep) time() time.Time {
+	return s.stepTime
+}
```

Comment:
nit: We typically arrange receiver functions for a type together with the type declaration.